### PR TITLE
Implement GitHub Actions testing

### PR DESCRIPTION
**Pull request by kmcelwee**

### CODE REVIEW

File: .github/workflows/unittests.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+      # Configure site
+      - name: Copy solr config
+        run: docker cp ${{ job.services.solr.id }}:/opt/solr/server/solr/$SOLR_CORE/conf tmp-solr-config
+      - name: run build_solr_schema to local schema
+        run: python manage.py build_solr_schema --configure-directory=tmp-solr-config
+      - name: upload configured schema
+        run: docker cp tmp-solr-config ${{ job.services.solr.id }}:/opt/solr/server/solr/$SOLR_CORE/conf
+      # - name: restart solr
+      #   run: docker exec --user solr ${{ job.services.solr.id }} /bin/bash -c "/opt/solr/bin/solr restart"
```
Comment (kmcelwee) [inline .github/workflows/unittests.yaml:68]: Try reloading solr core from solr CLI

Comment (kmcelwee) [reply .github/workflows/unittests.yaml:68]: Or create a new configset.

https://stackoverflow.com/questions/11540434/reload-solr-core-with-curl

File: .github/workflows/unittests.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      solr:
+        image: solr:6.6
+        ports:
+          - 8983:8983
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          # This version is necessary for CI to run, despite the site being on python 3.5
+          python-version: 3.7.6
```
Comment (kmcelwee) [inline .github/workflows/unittests.yaml:32]: Try 3.6

File: .github/workflows/unittests.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
+name: unit tests
+
+on:
+  push: # run on every push or PR to any branch
+  # pull_request:
```
Comment (kmcelwee) [inline .github/workflows/unittests.yaml:5]:
```suggestion
  pull_request:
```

File: .github/workflows/unittests.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      # copy core config to local machine and remove from docker vm
+      # use haystack to customize the config, copy it back, and reload the core
+      - name: Setup solr core
+        run: |
+          docker exec ${{ job.services.solr.id }} /bin/bash -c "/opt/solr/bin/solr create -c $SOLR_CORE -n basic_config"
+          docker cp ${{ job.services.solr.id }}:/opt/solr/server/solr/derrida/conf /tmp/derrida-conf
+          python manage.py build_solr_schema --configure-directory=/tmp/derrida-conf
+          docker exec ${{ job.services.solr.id }} /bin/bash -c "rm -rf /opt/solr/server/solr/derrida/conf"
+          docker cp /tmp/derrida-conf ${{ job.services.solr.id }}:/opt/solr/server/solr/derrida/conf
+          python manage.py build_solr_schema --configure-directory=/tmp/derrida-conf  --reload-core $SOLR_CORE
```
Comment (kmcelwee) [inline .github/workflows/unittests.yaml:58]: I guess the thing I couldn't figure out was to remove the existing config, and then you can reload the core from haystack, not the docker container?

**Review by kmcelwee — COMMENTED** [review]
THANK YOU.

File: .github/workflows/unittests.yaml (hunk 4 above)
Comment (rlskoeser) [reply .github/workflows/unittests.yaml:58]: yeah, it took a little bit of local testing for me to figure out a working solution — the basic config uses `managed-schema` but haystack still uses old-style `schema.xml`; if you pull it down haystack renames the managed one, but copying it back up without removing it leaves in place

We could have done the curl command to reload the container, but I did the lazy thing and just used haystack to regenerate the schema again and then run the reload command.

**Pull request merged**